File: erp_api/ERP.API/Responses/ApiResponse.cs
using System.Text.Json.Serialization;

namespace ERP.API.Responses;

public class ApiResponse<T>
{
    [JsonPropertyName("success")]
    public bool Success { get; set; }
    public T? Data { get; set; }
    public string Message { get; set; } = string.Empty;
    public List<string> Errors { get; set; } = new();
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;

    public static ApiResponse<T> SuccessResponse(T data, string message = "")
    {
        return new ApiResponse<T>
        {
            Success = true,
            Data = data,
            Message = message
        };
    }

    public static ApiResponse<object> ErrorResponse(string message, List<string>? errors = null)
    {
        return new ApiResponse<object>
        {
            Success = false,
            Message = message,
            Errors = errors ?? new List<string>()
        };
    }
}

File: erp_flutter/lib/core/models/api_response.dart
class ApiResponse<T> {
  final bool success;
  final T? data;
  final String message;
  final List<String> errors;
  final DateTime timestamp;

  ApiResponse({
    required this.success,
    this.data,
    required this.message,
    this.errors = const [],
    DateTime? timestamp,
  }) : timestamp = timestamp ?? DateTime.now();

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic) fromJsonT,
  ) {
    return ApiResponse(
      success: json['success'] ?? false,
      data: json['data'] != null ? fromJsonT(json['data']) : null,
      message: json['message'] ?? '',
      errors: (json['errors'] as List?)?.map((e) => e.toString()).toList() ?? [],
      timestamp: json['timestamp'] != null 
          ? DateTime.parse(json['timestamp']) 
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'success': success,
      'data': data,
      'message': message,
      'errors': errors,
      'timestamp': timestamp.toIso8601String(),
    };
  }
}
